examples/async_client.rs

```diff
@@ -0,0 +1,14 @@
+use futures::io::{AsyncReadExt, AsyncWriteExt};
+use interprocess::nonblocking::local_socket::*;
+
+#[tokio::main]
+async fn main() {
+    // Replace the path as necessary on Windows.
+    let mut conn = LocalSocketStream::connect("/tmp/example.sock")
+        .await
+        .unwrap();
+    conn.write_all(b"Hello from client!").await.unwrap();
+    let mut buffer = String::new();
+    conn.read_to_string(&mut buffer).await.unwrap();
+    println!("Server answered: {}", buffer);
+}
```

Comment:
```suggestion
use futures::io::{BufReader, AsyncBufReadExt, AsyncWriteExt};
use interprocess::nonblocking::local_socket::*;

#[tokio::main]
async fn main() {
    // Replace the path as necessary on Windows.
    let mut conn = LocalSocketStream::connect("/tmp/example.sock")
        .await
        .unwrap();
    conn.write_all(b"Hello from client!\n").await.unwrap();
    let mut conn = BufReader::new(conn);
    let mut buffer = String::new();
    conn.read_line(&mut buffer).await.unwrap();
    println!("Server answered: {}", buffer);
}
```
Trying to read until EOF with bidirectional sockets is always most likely an error, since that'd deadlock two processes. This is a replacement with the async `BufReader`, which will only read one line.